### PR TITLE
Add comparisons service

### PR DESCRIPTION
**Pull request by ptrang127**
Basic comparisons service to update comparisons
Update seed.sql to add unique constraints on subject ids.
Add wins and losses.
Reordering the fields on our models.

### CODE REVIEW

File: src/models/Comparisons.ts [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
   firstSubjectId?: number,
   secondSubjectId?: number,
   firstSubjectVotes?: number,
   secondSubjectVotes?: number,
-  id?: number, // id last cause usually set by db
 ): Comparison {
   return {
-    id: (id ?? -1),
-    category: (category ?? Categories.new()),
+    id: (id ?? undefined),
```
Comment (ptrang127) [inline src/models/Comparisons.ts:35]: DB ids can be undefined (record not created)

File: src/routes/ComparisonsRouter.ts [hunk 2]
```diff
@@ -1,20 +1,29 @@
...
 import { Comparison } from '@src/models/Comparisons';
+import ComparisonsService from '@src/services/ComparisonsService';
 
 
 const comparisonRouter = express.Router();
 
 // Define routes
 comparisonRouter.get('/', async function (req, res) {
-  const comparisons: Comparison[] = await pg('comparisons').select('*');
+  const comparisons: Comparison[] = await ComparisonsService.getAllComparisons();
```
Comment (KhoaLeTea) [inline src/routes/ComparisonsRouter.ts]: I imagine the UI flow is that the client presents one random comparison at a time. So they would probably call this endpoint. It sounds like a lot to return all comparisons in one API call.

File: src/routes/ComparisonsRouter.ts [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
   res.send(comparisons);
 });
 
 comparisonRouter.get('/:id', async function (req, res) {
```
Comment (KhoaLeTea) [inline src/routes/ComparisonsRouter.ts]: Would the UI ever call this endpoint? Idk the case when a client has a comparison ID and wants to look up just one comparison.

I imagined that the client for each new comparison just calls /comparison and it returns a random comparison between two subjects. That call alone should have all the information we need to display the comparison right?

File: src/routes/ComparisonsRouter.ts [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+  const comparisonId = parseInt(req.params.id);
+  const comparison: Comparison = await ComparisonsService.getComparisonById(comparisonId);
   res.send(comparison);
 });
 
+comparisonRouter.post('/:firstId/:secondId/:voteId', async function (req, res) {
+  const firstId = parseInt(req.params.firstId);
+  const secondId = parseInt(req.params.secondId);
+  const voteId = parseInt(req.params.voteId);
+  const comparison: Comparison = await ComparisonsService.incrementComparisonByFirstSubjectIdAndSecondSubjectId(firstId, secondId, voteId);
```
Comment (KhoaLeTea) [inline src/routes/ComparisonsRouter.ts:20]: This endpoint needs to update the wins and losses for each subject too right?

Or did you plan on adding that in a future PR?

File: src/routes/ComparisonsRouter.ts (hunk 2 above)
Comment (ptrang127) [reply src/routes/ComparisonsRouter.ts]: Agreed, we probably don't need this endpoint to get all comparisons.

For the UI flow, I'm imagining the following:
1. User selects a category
2. Based on the category, the backend fetches a list of subjects (get Subjects by category Id)
3. We can either return a list of tuple subjects back to the UI, or all of the subjects.
4. UI prompts users to select a subject. 
5. User votes and then we return the comparison object.

File: src/routes/ComparisonsRouter.ts (hunk 3 above)
Comment (ptrang127) [reply src/routes/ComparisonsRouter.ts]: Will remove this.

File: src/routes/ComparisonsRouter.ts (hunk 4 above)
Comment (ptrang127) [reply src/routes/ComparisonsRouter.ts:20]: Yeah that hasn't been added. We can use the `voteId` and call our SubjectsService and increment the score.

That will be a part of the Subjects Service PR

**Review by KhoaLeTea — APPROVED** [review]

**Pull request merged**